### DISCUSSION TITLE
build target name & libfile.c3l

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [c3lang/c3c#1694] [General]
Tested under win10 64 bit:
A. When a c3 source file with name file01.c3 and  there is no ``` module file01;``` in the file,
```c3c compile file01.c3```
 will produce a file01.exe; but when the module name is mod01 ,that is
```
moudle mod01;
```
the produced exe filename will be mod01.exe other than file01.exe.
Is this a bit weird?

B. Previously per my experience,under windows let's say if the sokol.c3l is a folder,we have to make it a zip file with the same name sokol.c3l and 
 move the original sokol.c3i folder to another place.Then compile the project which depends on lib sokol should be :
``` 
c3c build app.c3 --lib sokol --libdir ./sokol
```
Today I found that now I don't have to make a zip file in advance before compiling the project. That is ,just using the sokol.c3l folder with the same compiling command should also work.

Is this the build system a change recently or I just misunderstand something?

Thanks in advance.

**Comment by lerno** [comment]
If you don't provide a `-o MyName`, C3 will try to infer the name of the exe. This is done by looking at the module containing the `main` function.

**Comment by lerno** [comment]
Regarding .c3l: it may either be a zip file or a folder. The latter is especially convenient when developing a library.

**Comment by lerno** [comment]
Just a gentle reminder that discussions ( https://github.com/c3lang/c3c/discussions ) is probably a better place to ask questions than filing an issue.
